src/test/java/core/basesyntax/service/operation/SupplyOperationTest.java

```diff
@@ -0,0 +1,56 @@
+package core.basesyntax.service.operation;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import java.util.HashMap;
+import java.util.Map;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class SupplyOperationTest {
+    private Map<String, Integer> fruitsStorage;
+
+    @BeforeEach
+    void setup() {
+        fruitsStorage = new HashMap<>();
+        fruitsStorage.put("banana", 20);
+        Storage.setFruits(fruitsStorage);
+    }
+
+    @Test
+    void testProcessWithTransaction_successful() {
+        SupplyOperation supplyOperation = new SupplyOperation();
+        FruitTransaction transaction = new FruitTransaction(FruitTransaction.Operation.SUPPLY,
+                "banana", 10);
+        assertDoesNotThrow(() -> supplyOperation.processWithTransaction(transaction));
+        assertEquals(30, Storage.getFruits().get("banana"));
+    }
+
+    @Test
+    void testProcessWithTransaction_nullTransaction() {
```

Comment:
```suggestion
    void testProcessWithTransaction_nullTransaction_throwsException() {
```
please uniform test names everywhere - methodUnderTest_conditions_result